### PR TITLE
Add Foreman client repositories

### PR DESCRIPTION
**Pull request by ehelms**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.19
* [ ] 1.18
* [ ] 1.17

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

**Comment by ekohl** [comment]
I assume we'll also be naming these `1.20` rather than `3.9` after branching since we're using the foreman name.

**Comment by ehelms** [comment]
That's the idea @ekohl -- Katello would be reduced to it's server repository only.

**Comment by ehelms** [comment]
Needs https://github.com/theforeman/tool_belt/pull/101

**Comment by jturel** [comment]
Is the goal here to preserve the katello client tooling package names but ship them in a broader, more generic repo for any and all foreman client packages?

**Comment by ehelms** [comment]
@jturel Yup!

**Comment by ekohl** [comment]
There's now a merge conflict but otherwise I think this looks good. We will need to ensure we can manage the repo so we need a `.repo` file in a `-release` RPM and support in forklift.

**Comment by ehelms** [comment]
This should help with these cases -- https://github.com/theforeman/foreman-infra/pull/770

**Comment by ehelms** [comment]
[test rpm]

File: package_manifest.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ packages:
     plugins_packages: {}
     katello_packages: {}
     katello_client_packages: {}
```
Comment (zjhuntin) [inline package_manifest.yaml]: Is this still needed if it's being eliminated later?

File: package_manifest.yaml [hunk 2]
```diff
@@ -358,46 +359,45 @@ katello_packages:
...
-      - f28-katello-client-nightly
+      - f28-foreman-client-nightly
       - f27-base
       - f27-updates
       - f27-puppet-5
-      - f27-katello-client-nightly
+      - f27-foreman-client-nightly
   hosts:
     katello-host-tools: {}
-    katello-repos: {}
```
Comment (zjhuntin) [inline package_manifest.yaml]: Why does this need to be removed?

Comment (ekohl) [reply package_manifest.yaml]: The section is being renamed to foreman client repos. I don't think it makes sense for the client tools to ship the katello repos (which should be named `katello-release` if it follows the naming convention), especially if katello-repos is now server-only.

**Comment by ehelms** [comment]
I updated this a bit to put back the Katello client aspect. I think a safer approach is to build out the Foreman client repository and keep existing Katello bits in tact until the new repositories are fully online.

**Comment by ehelms** [comment]
@ekohl @zjhuntin @komidore64 I re-factored this to just add in the structure for the new tags but not mess with the old existing Katello tags. I think it would be safer to make that change once the entire client ecosystem is setup.

File: rel-eng/tito.props [hunk 3]
```diff
@@ -401,6 +401,29 @@ whitelist = ansiblerole-insights-client
   rubygem-smart_proxy_salt
   rubygem-smart_proxy_spacewalk
 
+[foreman-client-nightly-rhel7]
+disttag = .el7
+scl = tfm
+whitelist = katello-host-tools
```
Comment (ekohl) [inline rel-eng/tito.props]: Should we keep the whitelists empty as well if we're not adding any packages yet?

File: rel-eng/tito.props [hunk 4]
```diff
@@ -401,6 +401,29 @@ whitelist = ansiblerole-insights-client
   rubygem-smart_proxy_salt
   rubygem-smart_proxy_spacewalk
 
+[foreman-client-nightly-rhel7]
+disttag = .el7
+scl = tfm
```
Comment (ekohl) [inline rel-eng/tito.props]: This is the only one with SCL. Should we also have a nonscl version?

**Comment by ehelms** [comment]
Cleaned up tito.props for the tags to start empty.

**Pull request merged**